NPE in maven

Trying to use Fray in an existing Maven project I get an NPE:

```
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.pastalab.fray.maven:fray-plugins-maven:0.6.7:prepare-fray (prepare-fray) on project channels: Execution prepare-fray of goal org.pastalab.fray.maven:fray-plugins-maven:0.6.7:prepare-fray failed.

Caused by: java.lang.NullPointerException
    at org.pastalab.fray.maven.PrepareFrayMojo.execute (PrepareFrayMojo.kt:39)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
```

I followed the README, adding the dependency and plugin. I've currently have a single test annotated with `@ConcurrencyTest`. I also downgraded junit from 6 to 5.11.3:

```
<dependency>
        <groupId>org.junit.jupiter</groupId>
        <artifactId>junit-jupiter</artifactId>
        <version>5.11.3</version>
      </dependency>
```

as this seems to be the version used by Fray. Checked with Java 25 & 21 (OpenJDK), same results.

--- a/plugins/maven/build.gradle.kts
+++ b/plugins/maven/build.gradle.kts
@@ -15,11 +15,11 @@ dependencies {
   testImplementation(platform("org.junit:junit-bom:5.10.0"))
   testImplementation("org.junit.jupiter:junit-jupiter")
 
-  compileOnly(project(":instrumentation:jdk"))
+  implementation(project(":instrumentation:jdk"))
   implementation(project(":plugins:base", configuration = "shadow"))
   implementation(project(":instrumentation:agent", configuration = "shadow"))
-  implementation("org.pastalab.fray:fray-jvmti-linux-x86-64:0.2.0")
-  implementation("org.pastalab.fray:fray-jvmti-macos-aarch64:0.2.0")
+  implementation("org.pastalab.fray:fray-jvmti-linux-x8664:0.6.7")
+  implementation("org.pastalab.fray:fray-jvmti-macos-aarch64:0.6.7")
 }
 
 tasks.test { useJUnitPlatform() }

--- a/plugins/maven/src/main/kotlin/org/pastalab/fray/maven/PrepareFrayMojo.kt
+++ b/plugins/maven/src/main/kotlin/org/pastalab/fray/maven/PrepareFrayMojo.kt
@@ -24,7 +24,7 @@ class PrepareFrayMojo : AbstractMojo() {
   @Parameter(property = "plugin.artifactMap", required = true, readonly = true)
   private val pluginArtifactMap: Map<String, Artifact>? = null
 
-  @Parameter(property = "plugin.jdkPath") private val jdkPath: String? = null
+  @Parameter(property = "plugin.jdkPath") private val originalJdkPath: String? = null
 
   @Parameter(property = "fray.workDir", defaultValue = "\${project.build.directory}/fray")
   private val destFile: File? = null
@@ -53,7 +53,7 @@ class PrepareFrayMojo : AbstractMojo() {
             File(jvmtiPath),
             getJvmtiJarFile(),
             destFile.absolutePath)
-    initializer.createInstrumentedJDK(FrayVersion.version, jdkPath)
+    initializer.createInstrumentedJDK(FrayVersion.version, originalJdkPath)
     initializer.createJVMTiRuntime()
 
     val oldValue = project!!.properties.getProperty("argLine") ?: ""
@@ -84,7 +84,8 @@ class PrepareFrayMojo : AbstractMojo() {
           else -> throw Exception("Unsupported OS")
         }
 
-    val arch = System.getProperty("os.arch")
+    val arch =
+        System.getProperty("os.arch").replace("-", "").let { if (it == "amd64") "x8664" else it }
     return pluginArtifactMap!!["org.pastalab.fray:fray-jvmti-$os-$arch"]!!.file
   }
 }

